### PR TITLE
Maintenance/580/add windows e2e tests

### PR DESCRIPTION
**Pull request by cliffle**

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -4,29 +4,69 @@ on:
...
+          run: npm test -- --runInBand --ci --bail
+        env:
+          FIREFOX_WD_URL: http://localhost:4444/wd/hub
+          CHROME_WD_URL: http://localhost:4445/wd/hub
+      - name: run integration tests
         uses: GabrielBB/xvfb-action@v1
         with:
-          run: bash ./.build/run_${{ runner.os }}.sh
-      - name: Run e2e tests
+          run: npm run test:it
```
Comment (svettwer) [inline .github/workflows/cd.yml:36]: I know it is not an explicit task of the issue but there are also integration test for windows. I think we should execute them as well.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -18,21 +18,66 @@ jobs:
...
         with:
-          run: bash ./.build/run_${{ runner.os }}.sh
+          run: npm test -- --runInBand --ci --bail
+        env:
+          FIREFOX_WD_URL: http://localhost:4444/wd/hub
+          CHROME_WD_URL: http://localhost:4445/wd/hub
+      - name: run integration tests
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: npm run test:it
```
Comment (svettwer) [inline .github/workflows/ci.yml:41]: see cd.yml remark

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -6,29 +6,69 @@ on:
...
+          run: npm test -- --runInBand --ci --bail
+        env:
+          FIREFOX_WD_URL: http://localhost:4444/wd/hub
+          CHROME_WD_URL: http://localhost:4445/wd/hub
+      - name: run integration tests
         uses: GabrielBB/xvfb-action@v1
         with:
-          run: bash ./.build/run_${{ runner.os }}.sh
-      - name: Run e2e tests
+          run: npm run test:it
```
Comment (svettwer) [inline .github/workflows/release.yml:38]: see cd.yml remark

**Review by svettwer — CHANGES_REQUESTED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (cliffle) [reply .github/workflows/cd.yml]: I thought about adding integration tests for windows before but haven't found integration tests for `sakuli-application` or `sakuli-environment`, which are os specific tests. That's why I didn't add them. I totally forgot about the other it-tests :see_no_evil:

File: .github/workflows/cd.yml [hunk 4]
```diff
@@ -4,29 +4,81 @@ on:
...
+          FIREFOX_WD_URL: http://localhost:4444/wd/hub
+          CHROME_WD_URL: http://localhost:4445/wd/hub
+      - name: run integration tests
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: npm run test:it
+      - name: Merge coverage report
+        run: |
+          npm run coverage:merge
+          npm run coverage:merge-report
```
Comment (svettwer) [inline .github/workflows/cd.yml:40]:
```suggestion
```
Sorry, just saw that. We do not need a coverage merge on cd as we don't send it to sonar cloud anyways.

File: .github/workflows/cd.yml [hunk 5]
```diff
@@ -4,29 +4,81 @@ on:
...
+          sudo apt-get update
+          sudo apt-get install -y gnome-calculator
+          docker pull selenium/standalone-chrome-debug
+          docker pull selenium/standalone-firefox-debug
+          docker run -d -p 4444:4444 selenium/standalone-firefox-debug
+          docker run -d -p 4445:4444 selenium/standalone-chrome-debug
       - run: npm ci
       - run: npm run build
-      - name: Run unit/integration tests
+      - run: npm run coverage:clean
```
Comment (svettwer) [inline .github/workflows/cd.yml:25]:
```suggestion
```
coverage not required on cd as we don't send it

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -6,29 +6,81 @@ on:
...
+          sudo apt-get update
+          sudo apt-get install -y gnome-calculator
+          docker pull selenium/standalone-chrome-debug
+          docker pull selenium/standalone-firefox-debug
+          docker run -d -p 4444:4444 selenium/standalone-firefox-debug
+          docker run -d -p 4445:4444 selenium/standalone-chrome-debug
       - run: npm ci
       - run: npm run build
-      - name: Run unit/integration tests
+      - run: npm run coverage:clean
```
Comment (svettwer) [inline .github/workflows/release.yml:27]:
```suggestion
```
See cd.yml

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -6,29 +6,81 @@ on:
...
+          FIREFOX_WD_URL: http://localhost:4444/wd/hub
+          CHROME_WD_URL: http://localhost:4445/wd/hub
+      - name: run integration tests
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: npm run test:it
+      - name: Merge coverage report
+        run: |
+          npm run coverage:merge
+          npm run coverage:merge-report
```
Comment (svettwer) [inline .github/workflows/release.yml:42]:
```suggestion
```
see cd.yml

**Review by svettwer — CHANGES_REQUESTED** [review]
Thx for adding the integration tests.
I'd love if it would be more clear that it-tests is only running on windows as the linux integration tests are executed during coverage generation. Unfortunately I've no idea how we could achieve that without building the software over and over again. 

In addition, I've found two remarks that I did not recognize at the first review. I added suggestions so I think this should be easy to fix. Sorry for that! :bow:

**Review by svettwer — APPROVED** [review]

**Comment by svettwer** [comment]
Good job :+1:

**Pull request merged**